### PR TITLE
Dynamically insert dialogue options

### PR DESCRIPTION
**Pull request by fleskesvor**
No need to merge, unless anyone else wants to play around with this.

For now, I figure we can just store our dictionary in globals/dialog_dialog.gd, since it's where we'll use it to insert dialogue options, and we have easy access to the global vm object here.

What this does for now, is just to insert two dialogue options into the root of the dialogue tree, which both make the player character say the word, and the dinosaur will just respond with a "?" (question mark).

Any actually appropriate dialogue options should probably be defined in esc scripts to avoid making the code unmanageable, so we should only append words which aren't already present I guess.

Also, we might want to shuffle or alphabetize the list of words to avoid it becoming too obvious which words don't actually do anything?

### CODE REVIEW

**Comment by Cheeseness** [comment]
> Also, we might want to shuffle or alphabetize the list of words to avoid it becoming too obvious which words don't actually do anything?

Maybe it'd be best to keep words in the order that they were "learned" (not "understood"), so that the location of new/old words is obvious to players?

**Comment by fleskesvor** [comment]
Yeah, that sounds like a good idea.

**Comment by fleskesvor** [comment]
This isn't everything from the spec, eg. learned words don't stay in the list, since I don't know what we'd do to add any meaningful conversation options for those yet.

What it does though, is let the player repeat a word on the next turn, and doing so sets a global, marking the word as learned. If the learned word is repeated by the dino, it is understood.

I'm not sure if this is good enough for now, and as you can see by looking at globals/dialog_dialog.gd and scenes/characters/test_dino/test_dino.esc, much of it is driven by .esc script logic.

To test:

* Select "Hello!" option
* Dinosaur replies with "*bruuuuugh*"

There's now a "*bruuuuugh*" option at the bottom, but select something else, and the option goes away.

* Select "Hello!" option again
* Dinosaur replies with "*bruuuuugh*"
* Select "*bruuuuugh*" (word is internally marked as learned, ie. greeting_learned flag is set)
* Select "Hello!" again, dinosaur replies, and you now realize this is a greeting - word is marked as understood

Now the "Hello!" option will be gone, and instead replaced with the "(greeting)" branch, which could have a whole separate dialogue branch under it.

File: globals/global_vm.gd [hunk 1]
```diff
@@ -558,6 +558,9 @@ func save():
 
 	ret.append("## Global flags\n\n")
 	for k in globals.keys():
+		# TODO: Implement serialization of dictionary
+		if typeof(globals[k]) == TYPE_DICTIONARY:
+			continue
```
Comment (fleskesvor) [inline globals/global_vm.gd]: This is a hack to work around the issue mentioned in #9.

**Comment by Cheeseness** [comment]
Great! I'll take a look at this stuff this morning

> This isn't everything from the spec, eg. learned words don't stay in the list, since I don't know what we'd do to add any meaningful conversation options for those yet.

I've made a start on dinosaur dialogue trees already. I figure we'll just make the dinosaur say [confused] in response to any dialogue option that's not part of the defined dialogue tree?

**Comment by Cheeseness** [comment]
OK. This is a good start. I think we need a little more work to get the implementation solid, but I'll merge this now and we can build on it from there

**Comment by fleskesvor** [comment]
> I figure we'll just make the dinosaur say [confused] in response to any dialogue option that's not part of the defined dialogue tree?

Ok, that should be easy to fix by adding the get_lines() function from a1794f3aeb66dbbce3603715b2a42938ec668f0f back in, but it needs to also support `is_false`, the way it supports `is_true`. Then we'll loop over the word list, present the player with a learning opportunity if the word was just heard or a throwaway/temp branch if the word was learned and not among the already  available options, translated or not.

**Comment by Cheeseness** [comment]
Yeah, I think approach that sounds good.

**Pull request merged**